src/engine/strat_engine/engine.rs

```diff
@@ -122,6 +141,42 @@ impl Engine for StratEngine {
     fn pools(&self) -> Vec<&Pool> {
         self.pools.into_iter().map(|x| x as &Pool).collect()
     }
+
+    fn get_eventable(&mut self) -> EngineResult<Option<Box<Eventable>>> {
```

Comment:
I'm not sure that Box in the actual return type is a good idea.

Reply:
We need to return a trait object here, so I think this is needed.